### PR TITLE
[DEV-4926] New Program Activity Count endpoint for agencies

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
Returned the count of distinct program account categories for an agency in a single fiscal year

**Technical details:**
- This code will be updated after Kirk's changes 
- Reorganized fiscal year helper functions and added tests

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket [DEV-4926](https://federal-spending-transparency.atlassian.net/browse/DEV-4926):
    - [ ] Link to this Pull-Request
    - [x] Performance evaluation of affected API (under 2s)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No matview or data changes necessary
```

### CODE REVIEW

File: usaspending_api/api_contracts/contracts/v2/agency/agency_id/program_activity/count.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
++ Request (application/json)
+    + Schema
+
+            {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "number"
+            }
+    + Parameters
+        + `agency_id`: 100 (required, number)
```
Comment (kbard) [inline usaspending_api/api_contracts/contracts/v2/agency/agency_id/program_activity/count.md:19]:
```suggestion
        + `agency_id`: 020 (required, number)
```
Having a valid `agency_id` here making Dredd testing a little easier.

File: usaspending_api/common/helpers/fiscal_year_helpers.py [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+
+from dateutil.relativedelta import relativedelta
+from fiscalyear import FiscalDate, FiscalQuarter, FiscalDateTime, datetime
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.common.helpers.generic_helper import validate_date, min_and_max_from_date_ranges
+
+logger = logging.getLogger(__name__)
+
+
+def current_fiscal_date() -> FiscalDate:
```
Comment (kbard) [inline usaspending_api/common/helpers/fiscal_year_helpers.py:11]:
```suggestion
def current_fiscal_date() -> FiscalDateTime:
```

File: usaspending_api/api_docs/markdown/endpoints.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ The currently available endpoints are listed in the following table.
 
 | Endpoint | Methods | Description |
 | -------- | ------- | ----------- |
-|[/api/v2/agency/test/](/api/v2/agency/test/)|GET| TEMPORARY endpoint for new Django App |
+|[/api/v2/agency/<AGENCY_ID\>/program_activity/count/](/api/v2/agency/100/program_activity/count/)|GET| Returns the count of unique Program Activity categories for the agency in a single fiscal year |
```
Comment (kbard) [inline usaspending_api/api_docs/markdown/endpoints.md:31]:
```suggestion
|[/api/v2/agency/<AGENCY_ID\>/program_activity/count/](/api/v2/agency/020/program_activity/count/)|GET| Returns the count of unique Program Activity categories for the agency in a single fiscal year |
```

Comment (tony-sappe) [reply usaspending_api/api_docs/markdown/endpoints.md]: Having issues with internet/github. last commit wasn't pushed. Sorry

File: usaspending_api/common/helpers/fiscal_year_helpers.py (hunk 2 above)
Comment (tony-sappe) [reply usaspending_api/common/helpers/fiscal_year_helpers.py]: thanks

**Pull request merged**